lib/transformHtml.js

```diff
@@ -38,41 +37,20 @@ function asStream(listener) {
   return pumpify.obj(htmlParser, esiParser);
 }
 
-function convert(buf, chunkSize) {
-  buf = Buffer.from(buf, "utf8");
-
-  const reader = new Readable();
-  reader.setEncoding("utf8");
-  const hwm = reader._readableState.highWaterMark;
-
-  if (!chunkSize || chunkSize < 1 || chunkSize > hwm) {
-    chunkSize = hwm;
-  }
-
-  const len = buf.length;
-  let start = 0;
-
-  reader._read = function () {
-    while (reader.push(buf.slice(start, (start += chunkSize)))) {
-      if (start >= len) {
-        reader.push(null);
-        break;
-      }
-    }
-  };
-  return reader;
+function convert(buf, chunkSize, encoding = "utf8") {
+  return Readable.from(buf, {encoding});
 }
 
-function createESIParser({onopentag, ontext, onclosetag}) {
+function createESIParser(evaluator) {
```

Comment:
Not really part of this PR but `ESIParser` class is only used here and it's scoped to this filter / evaluator adapter. Feels weird. Should this belong to the class itself?